CMSgov/easi-app
# Notify requester and TRB admins when TRB request form is submitted

# EASI-2443

## Changes and Description

- When a TRB request form is submitted, send notification emails to the requester and to the TRB inbox.
- Introduces a new config variable, `TRB_EMAIL`, for the TRB inbox address.

## Email Appearance

These are screenshots of how Mailcatcher renders the emails:
![trb request submission notification - admins](https://user-images.githubusercontent.com/92891039/204862845-e0947831-3b4a-483e-8543-f1777672d030.PNG)
![trb request submission notification - requester](https://user-images.githubusercontent.com/92891039/204862860-c816da71-9540-47bd-8107-e626b72ab841.PNG)

## How to test this change

Run the following GraphQL mutation to create a TRB request:
```gql
mutation {
  createTRBRequest(requestType: NEED_HELP) {
    id
    type
    form {
      id
      status
      trbRequestId
      component
      needsAssistanceWith
      hasSolutionInMind
      proposedSolution
      whereInProcess
      whereInProcessOther
      hasExpectedStartEndDates
      expectedStartDate
      expectedEndDate
      collabGroups
      collabDateSecurity
      collabDateEnterpriseArchitecture
      collabDateCloud
      collabDatePrivacyAdvisor
      collabDateGovernanceReviewBoard
      collabDateOther
      collabGroupOther
      createdBy
      createdAt
      modifiedBy
      modifiedAt
    }
  }
}
```

Note down the returned request ID for use in the following steps.

Run the following GraphQL mutation to create an _unsubmitted_ form, then check Mailcatcher; emails should _not_ have been sent:

```gql
mutation createUnsubmittedForm {
  updateTRBRequestForm(input: {
    isSubmitted: false,
    trbRequestId: "trbRequestIdFromAbove",
    needsAssistanceWith: "Some of the things",
    hasSolutionInMind: true,
    proposedSolution: "It's like Tinder, but for cats",
    whereInProcess: CONTRACTING_WORK_HAS_STARTED,
    whereInProcessOther: "Fixing Wifi",
    hasExpectedStartEndDates: true,
    expectedStartDate:"2022-11-30T03:29:56.901Z",
    expectedEndDate:"2050-11-30T03:29:56.901Z",
    collabGroups: [SECURITY, CLOUD, PRIVACY_ADVISOR],
    collabDateSecurity: "2021-10-20T03:29:56.901Z",
    collabDateEnterpriseArchitecture: "2021-10-21T03:29:56.901Z",
    collabDateCloud: "2021-10-22T03:29:56.901Z",
    collabDatePrivacyAdvisor: "2021-10-23T03:29:56.901Z",
    collabDateGovernanceReviewBoard: "2021-10-24T03:29:56.901Z",
    collabDateOther: "2021-10-25T03:29:56.901Z",
    collabGroupOther: "2021-10-26T03:29:56.901Z",
    subjectAreaTechnicalReferenceArchitecture: [CMS_TRA_BUSINESS_RULES],
    subjectAreaNetworkAndSecurity: [GENERAL_NETWORK_AND_SECURITY_SERVICES_INFORMATION],
    subjectAreaCloudAndInfrastructure: [DISASTER_RECOVERY],
    subjectAreaApplicationDevelopment: [OTHER],
    subjectAreaDataAndDataManagement: [DATA_WAREHOUSING],
    subjectAreaGovernmentProcessesAndPolicies: [OTHER_AVAILABLE_TRB_SERVICES],
    subjectAreaOtherTechnicalTopics: [ASSISTANCE_WITH_SYSTEM_CONCEPT_DEVELOPMENT],
    component: "Taco Truck",
  }) {
    status
    trbRequestId
    component
    needsAssistanceWith
    hasSolutionInMind
    proposedSolution
    whereInProcess
    whereInProcessOther
    hasExpectedStartEndDates
    expectedStartDate
    expectedEndDate
    collabGroups
    collabDateSecurity
    collabDateEnterpriseArchitecture
    collabDateCloud
    collabDatePrivacyAdvisor
    collabDateGovernanceReviewBoard
    collabDateOther
    collabGroupOther
    subjectAreaTechnicalReferenceArchitecture
    subjectAreaNetworkAndSecurity
    subjectAreaCloudAndInfrastructure
    subjectAreaApplicationDevelopment
    subjectAreaDataAndDataManagement
    subjectAreaGovernmentProcessesAndPolicies
    subjectAreaOtherTechnicalTopics
    createdBy
    createdAt
    modifiedBy
    modifiedAt
  }
}
```

Run the following GraphQL mutation to submit the created form, then check MailCatcher - emails should have been sent to the requester and admin:

```gql
mutation submitExistingForm {
  updateTRBRequestForm(input: {
    isSubmitted: true,
    trbRequestId: "trbRequestIdFromAbove",
  }) {
    status
  }
}
```

## Notes
- The styles for the emails are a mix of CSS copied from the Figma mockups (linked from the Jira ticket), [Adam's advice on Slack](https://oddball.slack.com/archives/C037ZNEQ7DH/p1669220502594669), and my own judgment (adjusting margins, not specifying any font in order to stay consistent with the rest of our emails).
